### PR TITLE
Ka 2019 10 dashboard project create

### PR DESCRIPTION
**Pull request by fuzzylogic2000**
depends on: https://github.com/liqd/adhocracy4/pull/411

still missing:
- only publish when all required fields are filled in
- all the module-logic to only show and take into account published modules
- styling

### CODE REVIEW

File: meinberlin/apps/dashboard/views.py [hunk 1]
```diff
@@ -89,6 +91,86 @@ def get_permission_object(self):
...
+        if 'referrer' in self.request.POST:
+            return self.request.POST['referrer']
+        elif 'HTTP_REFERER' in self.request.META:
+            return self.request.META['HTTP_REFERER']
+
+        return reverse('a4dashboard:project-edit', kwargs={
+            'project_slug': self.project.slug
+        })
+
+    def publish_module(self):
```
Comment (MagdaN) [inline meinberlin/apps/dashboard/views.py]: the publish and unpublish function are pretty similar, maybe they can be combined somehow

Comment (fuzzylogic2000) [reply meinberlin/apps/dashboard/views.py]: Yeah, the function could know, which it gets?! But I think, it wouldn't be more readable, when we then add three more if-statements and only have two lines out of if-statements.

Comment (MagdaN) [reply meinberlin/apps/dashboard/views.py]: fair enough

File: meinberlin/apps/dashboard/views.py [hunk 2]
```diff
@@ -49,8 +52,9 @@ def post(self, request, *args, **kwargs):
         project = self.get_object()
         module = module_models.Module(
             name=self.blueprint.title,
-            weight=len(project.modules) + 1,
+            weight=max([module.weight for module in project.modules]) + 1,
```
Comment (MagdaN) [inline meinberlin/apps/dashboard/views.py]: could you also work with a value_list here? As there won't be a lot of modules, using a for loop should not be a problem, but if possible, I guess, it is a bit better to try to use DB functionalities.

File: meinberlin/apps/dashboard/views.py [hunk 3]
```diff
@@ -88,6 +92,119 @@ def get_permission_object(self):
...
+        kwargs['creator'] = self.request.user
+        return kwargs
+
+    def get_success_url(self):
+        return reverse('a4dashboard:project-edit',
+                       kwargs={'project_slug': self.object.slug})
+
+    def form_valid(self, form):
+        response = super().form_valid(form)
+        signals.project_created.send(sender=None,
```
Comment (MagdaN) [inline meinberlin/apps/dashboard/views.py]: Can we be always sure at this point that the creation was successful?

**Pull request merged**